src/utils/state.sh

```diff
@@ -1,21 +1,37 @@
+# The location where the states of the desktops are stored.
 DESKTOP_STATE="/tmp/bsp-layout.state/desktops";
 
-# (Data ->) :: Key -> Value -> Data
-append_option() { sed "/^$1:/d"; echo "$1:$2"; }
+# (|Dict[str, str]|, str, str) -> Dict(str)
```

Comment:
Since everything in bash is basically a string, having `str` everywhere doesn't add any value. I think instead, we should maintain a label for it so it's clear what each argument is expected to be.

```suggestion
# (|Dict[key, value]|, key, value) -> Dict[key, value]
```

Also,the way to describe pipes looks a little muddy. What do you think about this instead?
```bash
# Dict[key, value] | (key, value) -> Dict[key, value]
```
It makes the intent clear that the value is being piped in.

Reply:
The pipe thingy should be addressed in a398df8 :yum: 

still working on the naming of the signatures :+1:

Reply:
And should be addressed in 8a1e1dd :+1: